### PR TITLE
Fix 'convert to tuple' analyzer.

### LINKED ISSUE #31746: 'Convert anonymous type to tuple' not working.
I cannot get the callback code AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer.AnalyzeSyntax to get called at all.

I've tried debugging why, but got utterly lost in the analyzer codepaths.  I would often be making progress only to hit code that would bail out due to state being set elsewhere.  but it wasn't clear what state was setting that.  After about an hour of rinse/repeat trying to just determine why this analyzer was not being called, i gave up.

@mavasani can you take a look and see why the callback is not getting hit?  The analyzer is simple.  It just does 'RegisterSyntaxAction' with SyntaxKind.AnonymousObjectCreationExpression.  And you shoudl be able to hit this trivially with an anonymous type as simple as `new { A = 1, B = 2 }`.  Thanks!

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Change to a SemanticSpan analysis since this is an analyzer that uses semantic models, even though it only registers for syntax node actions.

Fixes #31746

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @mavasani

File: src/Features/Core/Portable/ConvertAnonymousTypeToTuple/AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ protected AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer()
         protected abstract int GetInitializerCount(TAnonymousObjectCreationExpressionSyntax anonymousType);
 
         public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
-            => DiagnosticAnalyzerCategory.SyntaxAnalysis;
```
Comment (mavasani) [inline src/Features/Core/Portable/ConvertAnonymousTypeToTuple/AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs]: Can we also rename `SyntaxAnalysis` to `SyntaxTreeAnalysis`?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/ConvertAnonymousTypeToTuple/AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs]: Can i do that in a separate PR? :)

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/ConvertAnonymousTypeToTuple/AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs]: (I'm working on that now.  But i'd like them to be separate.

Comment (mavasani) [reply src/Features/Core/Portable/ConvertAnonymousTypeToTuple/AbstractConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs]: Certainly.

**Comment by mavasani** [comment]
Tagging @jinujoseph for approval

**Comment by genlu** [comment]
@CyrusNajmabadi Why this isn't caught by tests?

**Comment by CyrusNajmabadi** [comment]
@genlu See: https://github.com/dotnet/roslyn/issues/31757

**Pull request merged**